interface/modules/custom_modules/oe-module-dorn/src/GlobalConfig.php

```diff
@@ -0,0 +1,152 @@
+        //         return false;
+        //     }
+        // }
+        return true;
+    }
+
+    public function getClientId()
+    {
+        return $this->getGlobalSetting(self::CONFIG_OPTION_CLIENTID);
+    }
+    public function getClientSecret()
+    {
+        $encryptedValue = $this->getGlobalSetting(self::CONFIG_OPTION_CLIENTSECRET);
+        return $this->cryptoGen->decryptStandard($encryptedValue);
+    }
+
+    public function getClientScope()
+    {
+        if ($this->getGlobalSetting(self::CONFIG_OPTION_ENVIRONMENT) == "S") {
+            return "https://stagingclaimrevcom.onmicrosoft.com/portal/api/.default";
+        } elseif ($this->getGlobalSetting(self::CONFIG_OPTION_ENVIRONMENT) == "D") {
+            return "https://claimrevportaldevelopment.onmicrosoft.com/labs.claimrev.com/.default";
+        }
+        return "https://portalclaimrev.onmicrosoft.com/portal/api/.default";
+    }
+
+    public function getClientAuthority()
+    {
+        if ($this->getGlobalSetting(self::CONFIG_OPTION_ENVIRONMENT) == "S") {
+            return "https://stagingclaimrevcom.b2clogin.com/stagingclaimrevcom.onmicrosoft.com/B2C_1_sign-in-service/oauth2/v2.0/token";
+        } elseif ($this->getGlobalSetting(self::CONFIG_OPTION_ENVIRONMENT) == "D") {
+            return "https://claimrevportaldevelopment.b2clogin.com/claimrevportaldevelopment.onmicrosoft.com/B2C_1_sign-in-service/oauth2/v2.0/token";
+        }
+        return "https://portalclaimrev.b2clogin.com/portalclaimrev.onmicrosoft.com/B2C_1_sign-in-service/oauth2/v2.0/token";
+    }
+
+    public function getApiServer()
+    {
+        return $this->getGlobalSetting(self::CONFIG_OPTION_API_URL);
+    }
+
+
+    public function getTextOption()
+    {
+        return $this->getGlobalSetting(self::CONFIG_OPTION_TEXT);
+    }
+
+    /**
+     * Returns our decrypted value if we have one, or false if the value could not be decrypted or is empty.
+     *
+     * @return bool|string
+     */
+    public function getEncryptedOption()
+    {
+        $encryptedValue = $this->getGlobalSetting(self::CONFIG_OPTION_ENCRYPTED);
+        return $this->cryptoGen->decryptStandard($encryptedValue);
+    }
+
+    public function getGlobalSetting($settingKey)
+    {
+        return $this->globalsArray[$settingKey] ?? null;
+    }
+
+    public function getGlobalSettingSectionConfiguration()
+    {
+        $settings = [
+            self::CONFIG_OPTION_ENVIRONMENT => [
+                'title' => 'ClaimRev Environment (P=Production)'
+                ,'description' => 'The system you connect to. P for production'
+                ,'type' => GlobalSetting::DATA_TYPE_TEXT
+                ,'default' => 'P'
+            ],
+            self::CONFIG_OPTION_API_URL => [
+                'title' => 'API URL'
+                ,'description' => 'The api system you to connect to'
+                ,'type' => GlobalSetting::DATA_TYPE_TEXT
+                ,'default' => 'production\url\when\known'
```

Comment:
this does not look right. seems like the server should already be known.

Reply:
Just a placeholder.